src/rov/flight_controller/include/flight_controller/flight_controller.hpp

```diff
@@ -0,0 +1,64 @@
+#ifndef FLIGHT_CONTROLLER_HPP
+#define FLIGHT_CONTROLLER_HPP
+
+#include "eigen3/Eigen/Core"
+
+#include <rclcpp/rclcpp.hpp>
+#include <sensor_msgs/msg/joy.hpp>
+#include <std_srvs/srv/empty.hpp>
+
+#include "Thruster.hpp"
+#include "rov_interfaces/msg/bno055_data.hpp"
+#include "rov_interfaces/msg/thruster_setpoints.hpp"
+#include "rov_interfaces/msg/pwm.hpp"
+
+#define NUM_THRUSTERS 8
+
+// TODO: ensure services are thread-safe
+class FlightController : public rclcpp::Node {
+public:
+    FlightController();
+
+private:
+    void registerThrusters();
+    void toggle_PID(const std_srvs::srv::Empty_Request::SharedPtr request, std_srvs::srv::Empty_Response::SharedPtr response);
+    void setpoint_callback(const rov_interfaces::msg::ThrusterSetpoints::SharedPtr setpoints);
+    void bno_callback(const rov_interfaces::msg::BNO055Data::SharedPtr bno_data);
+    void updateSimple();
+    void updatePID();
+    void normalizethrottles(Eigen::Matrix<double,NUM_THRUSTERS,1>* throttles);
+
+    rclcpp::Subscription<rov_interfaces::msg::ThrusterSetpoints>::SharedPtr thruster_setpoint_subscription;
+    rclcpp::Subscription<rov_interfaces::msg::BNO055Data>::SharedPtr bno_data_subscription;
+    rclcpp::Publisher<rov_interfaces::msg::PWM>::SharedPtr _publisher;
```

Comment:
Publisher of what? Name your duds! Professor Baldwin just released V20 of the coding standards document and you're going to be docked points!!